### PR TITLE
Add volk_8{u,i}_convert_{32,64}f

### PR DESCRIPTION
**Pull request by xloem**
based on volk_8u_s32f_convert_32f

Adds 4 kernels to do basic conversion from 8-bit data to floating point.

### CODE REVIEW

File: kernels/volk/volk_8i_convert_32f.h [hunk 1]
```diff
@@ -0,0 +1,306 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2012, 2014 Free Software Foundation, Inc.
```
Comment (noc0lour) [inline kernels/volk/volk_8i_convert_32f.h:3]: just 2019 in this case

File: kernels/volk/volk_8i_convert_32f.h [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+ * \li num_points: The number of values.
+ *
+ * \b Outputs
+ * \li outputVector: The output 32-bit floats.
+ *
+ * \b Example
+ * \code
+ * int N = 10000;
+ *
+ * <FIXME>
```
Comment (noc0lour) [inline kernels/volk/volk_8i_convert_32f.h:47]: Probably provide a small example how to use the kernel.

**Review by noc0lour — CHANGES_REQUESTED** [review]
Some minor things, but the code looks fine. 
If you don't have a copyright assignment yet please contact bhilburn at gnuradio.org to get one for this and future contributions.

**Comment by jdemel** [comment]
It seems like this PR is still mergeable. I'd like to merge it but it seems like we are still waiting for a CLA. Is there any progress in that direction?

**Comment by jdemel** [comment]
@xloem  Is there any update on the CLA?
We start to see merge conflicts and I'd be more than happy to avoid them by merging your PRs.
If you could do a rebase on current master, that would be great too. It'd trigger more CI as well.

**Comment by jdemel** [comment]
@xloem  I assume there's still no update on your CLA?

Anyways, we adopted a new system: DCO. Thus, you only need to sign off your commits. `git commit -s` for new commits. The details are explained in `DCO.txt`. With `git commit -s` you add a `Signed off by ...` line to the end of your commit message. You can add

I got this approach to sign previous commits from [stackoverflow](https://stackoverflow.com/a/58599218)
```
git rebase --exec 'git commit --amend --no-edit -n -s' -i HEAD~1
```
I updated the option `-s` to lower case. Also, `HEAD~1` should refer to your one commit. It would be great if you could rebase your PR onto current master. If you fix the 2 comments @noc0lour had that would be great too. With DCO you can just put your name in the copyright line.

**Comment by xloem** [comment]
Hey, I am so sorry, I was working on this years ago during a traumatic flashback to handle paranoia.  I didn't know I had left it open.  I added this so rtl-sdr devices would work more smoothly.

I'm still pretty surprised that gnuradio is using these kind of older approaches to speeding data up.  Everybody is using gpus nowadays.  Is there a reference to work towards adding GPU integration anywhere?

EDIT:
I was actually brainwashed and enslaved not to support community efforts, and it can be very hard to get in closer contact with people working on them.  I'll stay on this and work through the flashbacks until I get the CLA thing addressed.

**Comment by jdemel** [comment]
@xloem no need for a CLA anymore. We decided to go with a DCO approach now. Just like GNU Radio. Just sign off you commit and we can merge it (given positive reviews).

VOLK does still focus on CPU optimization. There's a middle ground between plain CPU implementations and GPU offload. Some situation where optimized CPU support is desirable are latency constraint situations where transferring data back and forth from a GPUs takes too much time. Your kernels are a good example for that actually. You might want to convert your data before processing on a CPU. It would be too costly to transfer them back and forth from a GPU.

**Comment by xloem** [comment]
re: cpu/gpu.  the latency concern depends on how costly the processing is of course.  it would make sense to share a vector acceleration library with many other projects, so that work and maintenance can be shared.  the biggest ones nowadays might be the machine learning ones, don't really know.

when i originally saw volk, i was excited to have a place to make for very very fast vector operations, but i was sad it didn't use c++, which i usually use the template features of for such things, constructing the code at compile-time for its specific use.  macros might reduce the code duplication in this library and provide for similar optimization.

I've found the DCO document and am working on handling it.

**Comment by xloem** [comment]
I'm working on this.  It is really hard for me.

I don't actually remember doing the work, but my body sure did.  I have volk checked out on this system, which is from the area I was living when I did the work.  I do not have a branch on this disk that reflects this pull request.  I am having severe amnesia about what is goin as I try to pursue this, making it hard to remember that I am still pursuing it.

I worked very hard for many years to contribute to rtl-sdr work.  I submitted multiple patches to the people who make the open source driver for rtl-sdr.  None of those patches have been accepted; they are still sitting on the mailing list.  They have since gotten a new maintainer, and I was told I must resubmit my patches for them to be reviewed, that they aren't reviewing the old patches.  It is very frightening and exhausting to continue this work.

```
[user@DESKTOP-E3P7TC0 volk]$ git branch
* master
[user@DESKTOP-E3P7TC0 volk]$ 
```
why do i have volk cloned if i didn't make any changes to it?  it must be needed to build gnuradio, and not present in redhat el7's package repository ...  the date on it is 2020, august 24 ... okay so it was on some earlier system ...  I don't remember checking out volk in 2020 ... but that makes sense if it was just incidental to build gnuradio ... i've tried to build gnuradio and set up with rtl-sdrs and failed so many times that those are like a dense blur for me filled with destroyed fragments ...  now i'm dissociating; nocolour has such a wonderufl name.  i have nocolor too.  it really gets you hurt to not pick a color, it really gets you hurt to resist the mafia families forcing all the people to fight each other ...

**Comment by xloem** [comment]
Okay, so here's what's up:

As technology advances we need open source radios, to stop implant hacking once it becomes a big dangerous thing.  The rtl-sdr is the cheapest radio out there, it should function smoothly and easily, so it can be powerful.

I don't have the capacity to address all the issues mentioned here.  I'm sorry.  It is too hard for me to work on radios right now, and this feature is too low-impact.  I'm working on open source device drivers for eeg equipment, right now.

Thank you so much for staying on this issue for so long, keeping it open for all these years, and being tolerant of my strange behavior.  If I address the things you mention, I'll reopen the PR or a new one.

Be well, and thank you again for your angel-like radio work.

**Comment by xloem** [comment]
The "reopen" button isn't working for me, but I have signed off that commit.
I copy-pasted the recommended command, the "exec" portion.  I'm trying to do further work but `-s` is actually `--strategy`.  I don't know why it would add a signed-off-by line.

**Comment by xloem** [comment]
The signed-off commit is 87538b68991b01cb7b4d32af172c5666c4bf92a7

**Comment by xloem** [comment]
THIS IS REALLY HARD.  WHAT IS GOING ON?
edit: apologies for this.  i am so sorry.  please disregard my comments here.  the files are not hard to make at all if you are familiar with this codebase.

Karl Semich: reminder: you want to work on transparent radio technology more than you want to work on eeg technology.  both involve logging, both of them.

**Comment by jdemel** [comment]
It seems like this is disabled by github in your specific case:
https://github.com/isaacs/github/issues/361
We can't reopen a PR after a force push.

Could you just open a new PR?

**Comment by xloem** [comment]
Thanks for pinging [edit: sorry, reminding] me.

**Pull request closed without merging**